lib/src/io.dart

```diff
@@ -484,7 +484,7 @@ final bool runningFromTest = Platform.environment.containsKey('_PUB_TESTING');
 /// Whether pub is running from within the Dart SDK, as opposed to from the Dart
 /// source repository.
 final bool _runningFromSdk =
-    !runningFromTest && Platform.script.path.endsWith('.snapshot');
+    !runningFromTest && Platform.script.path.endsWith('.snapshot.dart2');
```

Comment:
I think this might be an unrelated "snapshot" postfix. This one is coming from how we snapshot pub in the SDK build right?

Reply:
Oops, yes, you're right. Fixed.